### ISSUE TITLE
Devices/entities unavailable. No photos. "Failed to unload"

### ISSUE THREAD

**Issue by sdholden28** [Daanoz/ha-google-photos#57]
## Version of the custom_component
v0.6.3 and v0.6.4

## Describe the bug
After updating HA to 2024.7.5, Google Photos integration will not load albums after HA restart. I have installed the beta, with no change. If I completely remove the integration and setup from scratch with new credentials, it will work until restart, and then the problem returns. This seems like a very similar problem that Alexa Media Player recently experienced, documented here: https://github.com/alandtse/alexa_media_player/issues/2402
If I reload the integration, it tells me that I must restart HA to complete the reload. If I do that, the issue simply returns.

## Debug log
Logger: homeassistant.config_entries
Source: config_entries.py:801
First occurred: 4:45:01 PM (1 occurrences)
Last logged: 4:45:01 PM

Error unloading entry ****@gmail.com for google_photos
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 801, in async_unload
    result = await component.async_unload_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/google_photos/__init__.py", line 94, in async_unload_entry
    for service_name in hass.services.async_services()[DOMAIN]:
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
KeyError: 'google_photos'

2024-08-08 13:50:40.425 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration google_photos which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-08-08 13:54:28.823 WARNING (MainThread) [homeassistant.components.calendar] Setup of calendar platform google is taking over 10 seconds.
2024-08-08 13:54:28.823 WARNING (MainThread) [homeassistant.components.camera] Setup of camera platform google_photos is taking over 10 seconds.
2024-08-08 13:54:28.823 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform google_photos is taking over 10 seconds.
2024-08-08 13:54:28.823 WARNING (MainThread) [homeassistant.components.select] Setup of select platform google_photos is taking over 10 seconds.
2024-08-08 13:54:29.293 ERROR (MainThread) [homeassistant.components.calendar] Setup of platform google is taking longer than 60 seconds. Startup will proceed without waiting any longer.
2024-08-08 13:54:29.293 ERROR (MainThread) [homeassistant.components.camera] Setup of platform google_photos is taking longer than 60 seconds. Startup will proceed without waiting any longer.
2024-08-08 13:54:29.293 ERROR (MainThread) [homeassistant.components.sensor] Setup of platform google_photos is taking longer than 60 seconds. Startup will proceed without waiting any longer.
2024-08-08 13:54:29.293 ERROR (MainThread) [homeassistant.components.select] Setup of platform google_photos is taking longer than 60 seconds. Startup will proceed without waiting any longer.
File "/config/custom_components/google_photos/coordinator.py", line 426, in _async_update_data
File "/config/custom_components/google_photos/coordinator.py", line 363, in update_data
File "/config/custom_components/google_photos/coordinator.py", line 528, in refresh_album_list
File "/config/custom_components/google_photos/coordinator.py", line 475, in _get_album_list
File "/config/custom_components/google_photos/api.py", line 57, in get_resource
File "/config/custom_components/google_photos/coordinator.py", line 416, in _async_update_data
File "/config/custom_components/google_photos/coordinator.py", line 428, in _async_update_data
File "/config/custom_components/google_photos/camera.py", line 48, in async_setup_entry
File "/config/custom_components/google_photos/coordinator.py", line 76, in get_coordinator
File "/config/custom_components/google_photos/coordinator.py", line 426, in _async_update_data
File "/config/custom_components/google_photos/coordinator.py", line 363, in update_data
File "/config/custom_components/google_photos/coordinator.py", line 528, in refresh_album_list
File "/config/custom_components/google_photos/coordinator.py", line 475, in _get_album_list
File "/config/custom_components/google_photos/api.py", line 57, in get_resource
File "/config/custom_components/google_photos/coordinator.py", line 416, in _async_update_data
File "/config/custom_components/google_photos/coordinator.py", line 428, in _async_update_data
File "/config/custom_components/google_photos/sensor.py", line 32, in async_setup_entry
File "/config/custom_components/google_photos/coordinator.py", line 67, in get_coordinator
File "/config/custom_components/google_photos/coordinator.py", line 426, in _async_update_data
File "/config/custom_components/google_photos/coordinator.py", line 363, in update_data
File "/config/custom_components/google_photos/coordinator.py", line 528, in refresh_album_list
File "/config/custom_components/google_photos/coordinator.py", line 475, in _get_album_list
File "/config/custom_components/google_photos/api.py", line 57, in get_resource
File "/config/custom_components/google_photos/coordinator.py", line 416, in _async_update_data
File "/config/custom_components/google_photos/coordinator.py", line 428, in _async_update_data
File "/config/custom_components/google_photos/select.py", line 34, in async_setup_entry
File "/config/custom_components/google_photos/coordinator.py", line 67, in get_coordinator
2024-08-08 16:45:01.221 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry ****@gmail.com for google_photos
File "/config/custom_components/google_photos/__init__.py", line 94, in async_unload_entry
KeyError: 'google_photos'

![image](https://github.com/user-attachments/assets/7b8d9c60-01de-49fb-aa40-45080cefae0d)

HA Info:
Core
2024.7.4

Supervisor
2024.08.0

Operating System
12.4

Frontend
20240710.0

**Comment by Daanoz** [comment]
Right, so I tried to reproduce this in 2024.7.4 & 2024.8.0 (I could not find 2024.7.5), but both seem to work fine. 

How many albums do you have?

**Comment by Daanoz** [comment]
I just noticed, one of your log messages:
```
2024-08-08 13:54:28.823 WARNING (MainThread) [homeassistant.components.calendar] Setup of calendar platform google is taking over 10 seconds.
```

Is about a different google platform integration, which also seems to have issues. Are you sure there aren't any connection issues to google?

**Comment by sdholden28** [comment]
My apologies, it was HA 2024.7.4. I did further testing and could not find any issues with connection to google or my google account. I removed the integration entirely and rebooted HA. I then setup the google photos integration successfully. 3 albums loaded photos without any trouble or errors. Then I rebooted HA, and the issue I've mentioned here returned. I repeated this process completely, 3 times, with the same results. Then, just for giggles, I updated HA from 2024.7.4 to 2024.8.1, and the problem has disappeared. The integration is working as expected, across reboots, without issue. I have no useful comment or info on the cause, but it appears it was resolved with some change between HA 2024.7.4 and 2024.8.1. This issue can be closed.
